packages/filesystem/src/browser/filesystem-save-resource-service.ts

```diff
@@ -25,9 +25,14 @@ import { FileDialogService } from './file-dialog';
 @injectable()
 export class FilesystemSaveResourceService extends SaveResourceService {
```

Comment:
Does our API allow to ever save to anything else but the file system? If not, why to we have this subclass?

Reply:
Since the `SaveResourceService` lives in the `@theia/core` package, it doesn't have access to services such as `FileService`. In general, the `Saveable` API is disk-agnostic, but the `Save As` functionality isn't.

Reply:
To me it seems that the partition of responsiblities does not really fit reality, leading to stuff like this and for example the empty "save as" implementation in core. But I do realize this is not what you're trying to do in this PR.